### PR TITLE
add unit tests for admin menu, and omit more controls

### PR DESCRIPTION
**Pull request by davidsbailey**
# Description

Follow-on to https://github.com/code-dot-org/curriculumbuilder/pull/195, which was merged without tests to unblock the PL team. This PR extracts some test helpers and adds test cases to cover the code changes in #195 . 

update: this PR also omits Get Code Studio Stage Details from the menu for users who do not have permission to edit the page.

## Testing story

Adds unit test cases for each unique way that we render the admin menu.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: curricula/tests/test_curricula_rendering.py [hunk 1]
```diff
@@ -87,30 +87,53 @@ def test_render_lesson(self):
...
+        self.assert_admin_menu('/test-curriculum/test-unit/', True)
+        self.assert_admin_menu('/pl-curriculum/pl-unit/', True)
+
+    def test_render_lesson_admin_menu(self):
+        self.assert_admin_menu('/test-curriculum/test-unit/1/', False)
+        self.assert_admin_menu('/test-curriculum/hoc-unit/1/', False)
+        self.assert_admin_menu('/pl-curriculum/pl-unit/1/', False)
+
+        self.add_permission(self.user, 'change_lesson')
+        self.add_permission(self.user, 'access_all_lessons')
```
Comment (dmcavoy) [inline curricula/tests/test_curricula_rendering.py]: Why do you need both change lesson and access all lessons? I thought access_all_lessons would give you the ability to change Lesson

Comment (davidsbailey) [reply curricula/tests/test_curricula_rendering.py]: https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/curricula/views.py#L258-L259
https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/lessons/models.py#L450-L452
https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/lessons/models.py#L406-L408

The way we have it set up right now, access_all_lessons is not sufficient.

**Comment by davidsbailey** [comment]
@dmcavoy per offline discussion, I've updated this PR to exclude code-studio pull-through from the menu for people who do not have permission to view the page. please take another look!

**Comment by dmcavoy** [comment]
👍 With the addition of the comment, we discussed for the tests.

**Pull request merged**